Plugins/Flow.Launcher.Plugin.Explorer/Search/EnvironmentVariables.cs

```diff
@@ -1,83 +1,88 @@
-using System;
+﻿using System;
 using System.Collections;
 using System.Collections.Generic;
 using System.IO;
-using System.Text;
+using System.Linq;
+using Flow.Launcher.Plugin.SharedCommands;
 
 namespace Flow.Launcher.Plugin.Explorer.Search
 {
     public static class EnvironmentVariables
     {
+        private static Dictionary<string, string> _envStringPaths = null;
+        private static Dictionary<string, string> EnvStringPaths
+        {
+            get
+            {
+                if (_envStringPaths == null)
+                {
+                    LoadEnvironmentStringPaths();
+                }
+                return _envStringPaths;
+            }
+        }
+
         internal static bool IsEnvironmentVariableSearch(string search)
         {
-            return search.StartsWith("%") 
+            return search.StartsWith("%")
                     && search != "%%"
-                    && !search.Contains("\\") &&
-                    LoadEnvironmentStringPaths().Count > 0;
+                    && !search.Contains('\\')
+                    && EnvStringPaths.Count > 0;
         }
 
-        internal static Dictionary<string, string> LoadEnvironmentStringPaths()
+        public static bool HasEnvironmentVar(string search)
         {
-            var envStringPaths = new Dictionary<string, string>(StringComparer.InvariantCultureIgnoreCase);
+            // "c:\foo %appdata%\" returns false
+            var splited = search.Split(Path.DirectorySeparatorChar);
+            return splited.Any(dir => dir.StartsWith('%') && 
+                                        dir.EndsWith('%') &&
```

Comment:
I don't think we need endswith? Some may like to write `%appdata%\flowlauncher\` balblabla. No need to enforce expansion to go deeper

Reply:
It matches spilt path (%appdata% and flowlauncher), not the whole path.

Reply:
oh sorry I read it wrongly thanks